pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotControllerPeriodicTaskRestletResource.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.helix.ClusterMessagingService;
+import org.apache.helix.Criteria;
+import org.apache.helix.InstanceType;
+import org.apache.pinot.common.messages.RunPeriodicTaskMessage;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.core.periodictask.PeriodicTaskScheduler;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Api(tags = Constants.PERIODIC_TASK_TAG)
+@Path("/periodictask")
+public class PinotControllerPeriodicTaskRestletResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotControllerPeriodicTaskRestletResource.class);
+
+  @Inject
+  PinotHelixResourceManager _pinotHelixResourceManager;
+
+  @Inject
+  PeriodicTaskScheduler _periodicTaskScheduler;
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/run")
+  @ApiOperation(value = "Run controller periodic task against the specified table. If no table name is specified, task will run against all tables.")
+  public boolean runPeriodicTask(
+      @ApiParam(value = "Periodic Task Name", required = true) @QueryParam("taskname") String periodicTaskName,
+      @ApiParam(value = "Table Name (with type)", required = false) @QueryParam("tablename") String tableName) {
+    if (!_periodicTaskScheduler.hasTask(periodicTaskName)) {
+      throw new WebApplicationException("Periodic task '" + periodicTaskName + "' not found.",
+          Response.Status.NOT_FOUND);
+    }
+
+    LOGGER.info("Sending periodic task execution message to all controllers for running task {} against {}.",
+        periodicTaskName, tableName != null ? tableName + " table" : "all tables");
+
+    // Create and send message to send to all controllers (including this one)
+    Criteria recipientCriteria = new Criteria();
+    recipientCriteria.setRecipientInstanceType(InstanceType.PARTICIPANT);
+    recipientCriteria.setInstanceName("%");
+    recipientCriteria.setSessionSpecific(true);
+    recipientCriteria.setResource(CommonConstants.Helix.LEAD_CONTROLLER_RESOURCE_NAME);
+    recipientCriteria.setSelfExcluded(false);
+    RunPeriodicTaskMessage runPeriodicTaskMessage = new RunPeriodicTaskMessage(periodicTaskName, tableName);
+
+    ClusterMessagingService clusterMessagingService =
+        _pinotHelixResourceManager.getHelixZkManager().getMessagingService();
+    int messageCount = clusterMessagingService.send(recipientCriteria, runPeriodicTaskMessage, null, -1);
+    LOGGER.info("Periodic task execution message sent to {} controllers.", messageCount);
+    return messageCount > 0;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/names")
+  @ApiOperation(value = "Get comma-delimited list of all available periodic task names.")
+  public ArrayList<String> getPeriodicTaskNames() {
+    ArrayList<String> list = new ArrayList<>();
+    list.add(org.apache.pinot.controller.validation.BrokerResourceValidationManager.class.getSimpleName());
```

Comment:
There is a variable called `_tasksWithValidInterval` in PeriodicTaskScheduler class, sth you can refer to instead of keeping a list of controller names here, which is hard to maintain.

Reply:
Good idea. Done.